Feature: adjustable interval between elements

alexobviously opened the pull request:
I needed this functionality for one of my projects so I added it. Made a PR because I thought you might want it in the main repo.
I added `interval` as a named parameter to the `intersperse` function (and extension method).
It does this: `intersperse(2, [0, 0, 0, 0, 0, 0], interval: 2); // [0, 0, 2, 0, 0, 2, 0, 0];`


Also an unrelated note just in case you didn't know: on the [pub.dev page](https://pub.dev/packages/intersperse) it doesn't list the github repo for this package.

alexobviously committed:
Interval parameter

modulovalue commented:
Hello Alex,

Thank you for this PR. Some comments:

* I would create a new function for this to not add complexity to the most common use case.
* I think I'd prefer to call it stride and add an offset for completeness i.e. see https://github.com/google/vector_math.dart/blob/527da5771eb7f1bd61b9e5fdb884891d46c3235d/lib/src/vector_math_lists/scalar_list_view.dart#L30
* This introduces some undefined behavior e.g. what should a user expect when `interval` is negative or zero?



> Also an unrelated note just in case you didn't know: on the pub.dev page it doesn't list the github repo for this package.

I'll include that with a future update.


There are some other changes that I wanted to make to this package such as interspersing lists in-place, but I don't have the time to work on that right now so I'll have to close this :(

alexobviously commented:
No problem. Like I said, I just modified this for a project that needed this use case (specifically: a row with a widget inserted every 2 elements), and we're using my fork anyway.

But I've made some changes if you want them. The main function now looks like this:
```dart
Iterable<T> intersperse<T>(T element, Iterable<T> iterable, {int stride = 1, int? offset}) sync* {
  if (stride < 1) throw ArgumentError.value(stride, 'stride', 'stride must be > 0');
  final iterator = iterable.iterator;
  final _offset = offset ?? stride;
  int pos = 0;
  while (iterator.moveNext()) {
    if (pos >= _offset && (pos - _offset) % stride == 0) yield element;
    yield iterator.current;
    pos++;
  }
}
```

* Renamed to stride (I just chose interval before because I wasn't aware of a convention)
* Throws an exception when stride < 1
* Added an offset parameter - I also thought this would be a nice addition (haven't added tests yet but it works)

> I would create a new function for this to not add complexity to the most common use case.

Well since they're named parameters, they don't change anything from the user's perspective if they're just using the default use case. But if you want to use this and rename it, feel free. Or I can rename it if you tell me what you'd prefer the name to be.

So anyway, if you want to merge this then you can reopen it and I'll push the latest commit. If not then don't worry :)
